exercises/pangram/pangram_test.go

```diff
@@ -0,0 +1,44 @@
+package pangram
+
+import (
+	"fmt"
+	"testing"
+)
+
+const targetTestVersion = 1
+
+type testCase struct {
+	input         string
+	expected      bool
+	failureReason string
+}
+
+var testCases = []testCase{
+	{"", false, "sentence empty"},
+	{"The quick brown fox jumps over the lazy dog", true, ""},
+	{"a quick movement of the enemy will jeopardize five gunboats", false, "missing character 'x'"},
+	{"the quick brown fish jumps over the lazy dog", false, "another missing character 'x'"},
+	{"the 1 quick brown fox jumps over the 2 lazy dogs", true, ""},
+	{"7h3 qu1ck brown fox jumps ov3r 7h3 lazy dog", false, "missing letters replaced by numbers"},
+	{"\"Five quacking Zephyrs jolt my wax bed.\"", true, ""},
+	{"Victor jagt zwölf Boxkämpfer quer über den großen Sylter Deich.", true, ""},
+	{"Широкая электрификация южных губерний даст мощный толчок подъёму сельского хозяйства.", false, "Panagram in alphabet other than ASCII"},
+}
+
+func TestTestVersion(t *testing.T) {
+	if testVersion != targetTestVersion {
+		t.Errorf("Found testVersion = %v, want %v.", testVersion, targetTestVersion)
+	}
+}
+
+func TestPangram(t *testing.T) {
+	for _, test := range testCases {
+		actual := isPangram(test.input)
+		if actual != test.expected {
+			t.Errorf("Pangram test [%s], expected [%s], actual [%s]", test.input, test.expected, actual)
+		}
+		if !test.expected {
+			fmt.Printf("Pangram test [%s], failure reason : %s\n", test.input, test.failureReason)
```

Comment:
Can you clarify what you intend? Now if the `isPangram` returns `false` when it should return `false`, this will print a `failure` reason even though the test has not failed, is that OK? Seems like it might be confusing to the student. Should this go inside the `if actual != test.expected` instead?

I also believe `Logf` messages *are* shown if tests fail, so it would have been OK to use `Logf`.